### PR TITLE
updates to the readme to use 1.0 release in docs

### PR DESCRIPTION
**Pull request by tsloughter-splunk**

### CODE REVIEW

**Review by theletterf — APPROVED** [review]

**Comment by theletterf** [comment]
Why are the e2e tests failing?

File: splunk-otel.gemspec [hunk 1]
```diff
@@ -28,7 +28,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency "opentelemetry-api", "~> 1.0"
   spec.add_dependency "opentelemetry-exporter-jaeger", ">= 0.20.1", "< 0.23.0"
   spec.add_dependency "opentelemetry-exporter-otlp", ">= 0.21", "< 0.25"
-  spec.add_dependency "opentelemetry-instrumentation-base", "~> 0.21.0"
+  spec.add_dependency "opentelemetry-instrumentation-base", "~> 0.21"
```
Comment (jtmal-signalfx) [inline splunk-otel.gemspec:31]: won't this do more harm than good? this lets consumers accidentally install mismatched packages

Comment (theletterf) [reply splunk-otel.gemspec:31]: Also wondering about the trailing `.0`. We have it in the docs for -all.

Comment (tsloughter-splunk) [reply splunk-otel.gemspec:31]: It should provide consumers more flexibility. It is basically the same as the jaeger and otlp dependencies above it, just without an upper limit that has to be bumped later.

Comment (jtmal-signalfx) [reply splunk-otel.gemspec:31]: if you're good I'm good!

**Pull request merged**